lib/DueDatePicker/DueDateHeader.js

```diff
@@ -27,38 +33,33 @@ const DueDateHeader = props => {
       'color-overdue': date < moment()
     });
 
-    return (
-      <div className="duedate__header">
-        Due date
-        <div className={classes}>
-          {day} <DueDateTime time={time} setTime={props.setTime} />
-        </div>
-        <DropdownMenu
-          className="duedate__remove"
-          caret={false}
-          type="link"
-          shouldDisplay
-          listClassName="dropdown-menu-right"
-          items={[
-            {
-              action: props.removeDueDate,
-              name: 'Remove due date',
-              linkType: 'danger'
-            }
-          ]}
-        >
-          <Icon name="menuDotted" />
-        </DropdownMenu>
+    label = (
+      <div className={classes}>
+        {day} <DueDateTime time={time} setTime={props.setTime} />
       </div>
     );
   }
 
+  const dropdownClasses = cx('duedate__remove', {
+    'duedate__remove--not-set': !props.dueDate
+  });
+
   return (
     <div className="duedate__header">
       Due date
-      <div className="duedate__header--date duedate__header--not-set">
-        No due date set
-      </div>
+      {label}
```

Comment:
This is more a stylistic preference, so feel free to ignore. It the label/header could be done like this:

```
const headerClasses = cx('duedate__header--date', {
    'duedate__header--date-not-set': !props.dueDate,
    'color-overdue': props.dueDate && moment(props.dueDate) < moment(),
  })

  const getDueDateHeader = () => {
    if (props.dueDate) {
      const date = moment(props.dueDate);
      const day = date
        .set({ hour: props.dueTime.hour(), minute: props.dueTime.minute() })
        .calendar(null, {
          sameDay: '[Today at]',
          nextDay: '[Tomorrow at]',
          nextWeek: 'dddd [at]',
          lastDay: '[Yesterday at]',
          lastWeek: '[Last] dddd [at]',
          sameElse: 'MMMM Do YYYY [at]'
        });
      const time = props.dueTime.format('LT');

      return <div className={headerClasses}>{day} <DueDateTime time={time} setTime={props.setTime} /></div>;
    }
    return <div className={headerClasses}>No due date set</div>;
  }

```

Then…

```
<div className="duedate__header">
      Due date
      {getDueDateHeader()}

```